dlib/serialize.h

```diff
@@ -983,6 +1224,144 @@ namespace dlib
 
 // ----------------------------------------------------------------------------------------
 
+    template <typename domain, typename range, typename hash, typename keyEqual, typename alloc>
+    void serialize (
+        const std::unordered_map<domain, range, hash, keyEqual, alloc>& item,
+        std::ostream& out
+    )
+    {
+        try
+        { 
+            const unsigned long size = static_cast<unsigned long>(item.size());
```

Comment:
Don't need this cast, just `serialize(item.size(), out)` is fine.

Reply:
Okidok. I was simply following what other areas of the code were doing in ```serialize.h```. I assumed there was maybe some subtle reason for doing this that you'd found.

Reply:
Do you want this change applied to all containers? It's not just unordered_map that casts to unsigned long...

Reply:
I'm going to assume yes

Reply:
Yeah, that's a good change.  I forgot those casts were even there.  I honestly don't remember what compiler it was that had problems with size_t.  It was probably visual studio though.  You can see lots of comments in dlib in front of bizarre code to the effect of "we do this silly thing to avoid a bug in visual studio".